include/os/linux/spl/sys/sysmacros.h

```diff
@@ -192,6 +192,12 @@ extern void spl_cleanup(void);
 #define	SET_ERROR(err) \
 	(__set_error(__FILE__, __func__, __LINE__, err), err)
 
+#if defined(_KERNEL)
```

Comment:
This guard shouldn't be needed, the header is only included in kernel context and `./lib/libspl/include/os/linux/sys/sysmacros.h` is used in user space.